odo/backends/aws.py

```diff
@@ -219,11 +220,31 @@ def anything_to_s3_text(s3, o, **kwargs):
     return append(s3, convert(Temp(s3.subtype), o, **kwargs), **kwargs)
 
 
+@contextmanager
+def start_multipart_upload_operation(s3):
+    try:
+        multipart_upload = s3.object.bucket.initiate_multipart_upload(s3.key)
+        yield multipart_upload
+    except Exception:
+        for part in multipart_upload:
+            s3.object.bucket.cancel_multipart_upload(part.key_name, part.id)
+        raise
+    else:
+        multipart_upload.complete_upload()
+
+
 @append.register(S3(JSONLines), (JSONLines, Temp(JSONLines)))
 @append.register(S3(JSON), (JSON, Temp(JSON)))
 @append.register(S3(CSV), (CSV, Temp(CSV)))
 @append.register(S3(TextFile), (TextFile, Temp(TextFile)))
-def append_text_to_s3(s3, data, **kwargs):
+def append_text_to_s3(s3, data, multipart=False, part_size=5 * 1 << 20, **kwargs):
+    if multipart:
+        with start_multipart_upload_operation(s3) as multipart_upload:
+            with open(data.path, 'rb') as f:
+                for part_number, part in enumerate(iter(curry(f.read, part_size), '')):
```

Comment:
can you make this `enumerate(..., start=1)`, then you don't need `part_number + 1`?